### PR TITLE
fix(test): Jest with ES Modules of react-dnd

### PR DESCRIPTION
**Pull request by dengfuping**
### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 👻 What's the background?

- Jest does not work well with ES Modules yet, so use the CommonJS builds of the react-dnd libraries after `react-dnd` updates to `v9.2.1`.
- ref: https://github.com/react-dnd/react-dnd/pull/1460

### 💡 Solution

- Configure Jest's `moduleNameMapper` settings for `react-dnd`:
```js
moduleNameMapper: {
    '^dnd-core$': 'dnd-core/dist/cjs',
    '^react-dnd$': 'react-dnd/dist/cjs',
    '^react-dnd-html5-backend$': 'react-dnd-html5-backend/dist/cjs',
    '^react-dnd-touch-backend$': 'react-dnd-touch-backend/dist/cjs',
    '^react-dnd-test-backend$': 'react-dnd-test-backend/dist/cjs',
    '^react-dnd-test-utils$': 'react-dnd-test-utils/dist/cjs',
  }
```

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### CODE REVIEW

**Comment by zombieJ** [comment]
这个是修 master ci 的吗？

**Comment by dengfuping** [comment]
@zombieJ 对的，react-dnd 升级导致 CI 挂了。

**Comment by zombieJ** [comment]
https://github.com/ant-design/ant-design/blob/b6c9903f319d67878494a4236c6b27b1b3ecfe0d/.jest.js#L5

这里可以删了

**Comment by dengfuping** [comment]
@zombieJ done.

File: .jest.js [hunk 1]
```diff
@@ -1,15 +1,20 @@
 const libDir = process.env.LIB_DIR;
 
-const transformIgnorePatterns = [
-  '/dist/',
-  'node_modules/(?!(_react-dnd|react-dnd|_dnd-core|dnd-core))[^/]+?/(?!(es|node_modules)/)', // Ignore modules without es dir
-];
+const transformIgnorePatterns = ['/dist/'];
```
Comment (zombieJ) [inline .jest.js]: 还原成这个：
https://github.com/ant-design/ant-design/blame/ae26f76d945c462b6315ebd4a3740a9fd61ca610/.jest.js#L5

Comment (dengfuping) [reply .jest.js]: - 已还原

**Comment by dengfuping** [comment]
CI passed ~

**Pull request merged**